### ISSUE TITLE
Assertion failed

### ISSUE THREAD

**Issue by x1314aq** [ninja-build/ninja#1971]
I have an error using latest `ninja` on Windows. It failed with following:

```
CMake Error:
  Running

   '/usr/local/bin/ninja.exe' '-C' '/cygdrive/c/Users/qwe/code/dsa/build' '-t' 'recompact'

  failed with:

   assertion "deps_data[i] < (int)nodes_.size()" failed: file "/cygdrive/c/Users/qwe/code/ninja-1.10.2/src/deps_log.cc", line 217, function: LoadStatus DepsLog::Load(const string&, State*, std::string*)

  Subprocess aborted
```

There are various files maybe you are interested in.

[ninja.zip](https://github.com/ninja-build/ninja/files/6495253/ninja.zip)

**Comment by x1314aq** [comment]
any updates?

**Comment by ZEPORATH** [comment]
any updates?

**Comment by jonesmz** [comment]
The reality of many open source projects is that there is not a centralized organization paying people to verify all the issues reported are issues, and then investigate and enact a fix.

While I won't say that it's not possible for some random internet stranger to see this issue and investigate a fix for it, the most likely outcome is that this gets ignored completely unless someone else encounters the exact issue and feels motivated to fix it, or the issue gets fixed by accident.

So no, no updates.

**Comment by jonesmz** [comment]
Does the latest version of ninja change the behavior you are seeing?

**Comment by rbrugo** [comment]
I'm having the same issue with ninja 1.12.1:
```
ninja: /usr/src/debug/ninja/ninja-1.12.1/src/deps_log.cc:217: LoadStatus DepsLog::Load(const std::string&, State*, std::string*): Assertion `deps_data[i] < (int)nodes_.size()' failed.
Subprocess aborted
```

**Comment by digit-google** [comment]
This was fixed recently by PR#2489
